packages/open-next/src/adapters/cache.ts

```diff
@@ -258,37 +238,44 @@ export default class S3Cache {
     }
     if (data?.kind === "ROUTE") {
       const { body, status, headers } = data;
-      await Promise.all([
-        this.putS3Object(key, "body", body),
-        this.putS3Object(key, "meta", JSON.stringify({ status, headers })),
-      ]);
+      this.putS3Object(
+        key,
+        "cache",
+        JSON.stringify({
+          type: "route",
+          body: body.toString("utf8"),
+          meta: {
+            status,
+            headers,
+          },
+        } as S3CachedFile),
+      );
     } else if (data?.kind === "PAGE") {
       const { html, pageData } = data;
       const isAppPath = typeof pageData === "string";
-      let metaPromise: Promise<PutObjectCommandOutput | void> =
-        Promise.resolve();
-      if (data.status || data.headers) {
-        metaPromise = this.putS3Object(
-          key,
-          "meta",
-          JSON.stringify({ status: data.status, headers: data.headers }),
-        );
-      }
-      await Promise.all([
-        this.putS3Object(key, "html", html),
-        this.putS3Object(
-          key,
-          isAppPath ? "rsc" : "json",
-          isAppPath ? pageData : JSON.stringify(pageData),
-        ),
-        metaPromise,
-      ]);
+      this.putS3Object(
+        key,
+        "cache",
+        JSON.stringify({
+          type: isAppPath ? "app" : "page",
+          html,
+          rsc: isAppPath ? pageData : undefined,
+          json: isAppPath ? undefined : pageData,
+          meta: { status: data.status, headers: data.headers },
+        } as S3CachedFile),
+      );
     } else if (data?.kind === "FETCH") {
       await this.putS3Object(key, "fetch", JSON.stringify(data));
     } else if (data?.kind === "REDIRECT") {
-      // delete potential page data if we're redirecting
-      await this.deleteS3Objects(key);
-      await this.putS3Object(key, "redirect", JSON.stringify(data));
+      // // delete potential page data if we're redirecting
```

Comment:
This comment may be confusing, since the command is `PUT` and not `DELETE`. Perhaps you can provide more context?

Reply:
My bad, i forgot to remove the comment